src/openforms/utils/helpers.py

```diff
@@ -0,0 +1,12 @@
+def truncate_str_if_needed(original_str: str, new_str: str, max_length: int) -> str:
+    """
+    Given the original string and the new string, return either the original one if
+    it doesn't exceed the given max length, or the new one truncated and by adding
+    the ellipsis to the original one.
+    """
+    if len(new_str) <= max_length:
+        return new_str
+
+    new_str_length = len(new_str) - len(original_str)
+    str_max_length = max_length - new_str_length
+    return original_str[: str_max_length - 1] + "\u2026" + new_str[-new_str_length:]
```

Comment:
I wonder if it doesn't make more sense to have a method that takes in the original string and the max length, then does something along the lines of:
```python
import textwrap

def copy_name(original_name, max_length):
     length_without_copy = max_length - 7
     return textwrap.shorten(original_name, width=length_without_copy, placeholder="\u2026") + " (copy)"
```
And this is used directly in src/openforms/forms/models/form_definition.py on lines 114/115?

(note that `shorten` breaks in between the words rather than in the words, but I don't think this is a problem for this case? :thinking: )

Reply:
As Sergei mentioned as well, we should avoid explicit numbers (the actual length of the additional str) that's why I did it this way. Additionally the length will be `7` ( " (copy)") in english but `8` in dutch (" (kopie)"), and this is the reason I did it more generic, like a function that could be used by anything. What is your opinion?

Reply:
Ahh makes sense!